calculator.js

```diff
@@ -1,5 +1,223 @@
+}
+
+const exampleMissingOp = {
+  num1: 3,
+  num2: 5,
+  op: null,
+}
+
+const exampleWrongSign = {
+  num1: 3,
+  num2: 5,
+  op: '(',
+}
+
+const exampleOneNonNum = {
+  num1: 'hotdog',
+  num2: 5,
+  op: '/',
+}
+
+const exampleTwoNonNum = {
+  num1: 'hotdog',
+  num2: 'elephant',
+  op: '*',
+}
+
+const exampleDivideByZero = {
+  num1: 5,
+  num2: 0,
+  op: 'divide',
+}
+
+/// CALCULATOR FUNCTIONALITY ///
+const operator = {
+  add: '+',
+  subtract: '-',
+  divide: '/',
+  multiply: '*',
+  exponent: '**',
+  modulo: '%',
+}
+
+function verifyNum(input) {
+  return isNaN(input) || !input
+}
+
+function printNum(data, result) {
+  console.log(`${data.num1} ${operator[data.op.toLowerCase()] || data.op} ${data.num2} = ${result}`)
+}
+
+function calculator(data) {
+  let sign = String(data.op)
+  sign = operator[sign.toLowerCase()] || sign
+  
+  if (verifyNum(data.num1) || verifyNum(data.num1)) {
+    console.log("Input provided is not a number")
+  }
+  else if (data.num2 === 0) {
+    console.log("Cannot divide by zero")
+  }
+  else if (sign === '+') {
+    printNum(data, data.num1 + data.num2)
+  } 
+  else if (sign === '-') {
+    printNum(data, data.num1 - data.num2)
+  } 
+  else if (sign === '*') {
+    printNum(data, data.num1 * data.num2)
+  } 
+  else if (sign === '/') {
+    printNum(data, data.num1 / data.num2)
+  } 
+  else if (sign === '%') {
+    printNum(data, data.num1 % data.num2)
+  } 
+  else if (sign === '**') {
+    printNum(data, data.num1 ** data.num2)
+  } 
+  else {
+    console.log("Operator not recognized")
+  }
+} 
+
+/// BASE CASE EXAMPLES ///
+
+console.log("*** ADDITION ***");
+calculator(exampleAdditionInput);
+calculator(exampleAdditionInputSign);
+console.log("*** SUBTRACTION ***");
+calculator(exampleSubtractInput);
+calculator(exampleSubtractInputSign);
+console.log("*** MULTIPLICATION ***");
+calculator(exampleMultiInput);
+calculator(exampleMultiInputSign);
+console.log("*** DIVIDE ***");
+calculator(exampleDivideInput);
+calculator(exampleDivideInputSign);
+console.log("*** MODULO ***");
+calculator(exampleModuloInput);
+calculator(exampleModuloInputSign);
+console.log("*** EXPONENT ***");
+calculator(exampleExponentInput);
+calculator(exampleExponentInputSign);
+
+
+/// EDGE CASE EXAMPLES ///
+
+console.log("*** MISSING NUMBER ***")
+calculator(exampleMissingNum);
+console.log("*** MISSING OPERATOR ***")
+calculator(exampleMissingOp);
+console.log("*** WRONG SIGN ***")
+calculator(exampleWrongSign);
+console.log("*** ADDING NON-NUMBER ***")
+calculator(exampleOneNonNum);
+calculator(exampleTwoNonNum);
+console.log("*** DIVIDE BY ZERO ***")
+calculator(exampleDivideByZero);
+
+console.log("*************************************");
+console.log("You can now enter in your own values to calculate:");
+
+// This line "loads" the prompt package and puts it into a variable we can use called prompt
+// Lines like this (that use require()) typically go on the top of the file
+const prompt = require('prompt');
```

Comment:
As you mentioned in your reflection questions, it would be good to at least separate the versions that do and don't take user input into different files.